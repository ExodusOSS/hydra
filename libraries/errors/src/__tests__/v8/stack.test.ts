import parseStackTrace from '../../stack.js'
import { extendedExpect } from '../setup.js'

function foo() {
  return new Error('boo')
}

it(`gets stack trace for v8`, () => {
  const err = foo()
  const stack = parseStackTrace(err)!
  const { stack: originalStack } = err
  expect(err.stack).toEqual(originalStack)
  expect(Array.isArray(stack)).toBe(true)

  const callSiteMatcher = {
    // all other props are matched exactly
    file: extendedExpect.filePath(),
    line: extendedExpect.number(),
    column: extendedExpect.number(),
  }

  expect({ ...stack }).toMatchSnapshot({
    ...Array.from({ length: stack.length }).fill(callSiteMatcher),
  })
})
