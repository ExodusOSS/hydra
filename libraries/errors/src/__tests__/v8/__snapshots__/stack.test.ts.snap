exports[`gets stack trace for v8 1`] = `
{
  "0": {
    "async": false,
    "column": number<>,
    "file": filePath<>,
    "function": "foo",
    "line": number<>,
    "method": null,
    "toplevel": true,
  },
  "1": {
    "async": false,
    "column": number<>,
    "file": filePath<>,
    "function": null,
    "line": number<>,
    "method": null,
    "toplevel": false,
  },
  "2": {
    "async": false,
    "column": number<>,
    "file": filePath<>,
    "function": "runInAsyncScope",
    "line": number<>,
    "method": "runInAsyncScope",
    "toplevel": false,
  },
  "3": {
    "async": false,
    "column": number<>,
    "file": filePath<>,
    "function": "run",
    "line": number<>,
    "method": "run",
    "toplevel": false,
  },
  "4": {
    "async": true,
    "column": number<>,
    "file": filePath<>,
    "function": "processPendingSubtests",
    "line": number<>,
    "method": "processPendingSubtests",
    "toplevel": false,
  },
}
`;
